packages/validator/src/validator.ts

```diff
@@ -366,12 +366,10 @@ export class Validator {
   private async fetchBeaconHealth(): Promise<BeaconHealth> {
     try {
       const {status: healthCode} = await this.api.node.getHealth();
-      // API always returns http status codes
-      // Need to find a way to return a custom enum type
-      if ((healthCode as unknown as routes.node.NodeHealth) === routes.node.NodeHealth.READY) return BeaconHealth.READY;
```

Comment:
@nazarhussain this seems to be a case where typed status codes might provide additional value as `getHealth` is the only api that returns the result as par of `.status` instead of `.value()`. But this seems disfunctional on unstable branch, requiring type hacks to even do assertions.

As you can see in the commit, those type casts are no longer required on this branch which seems like a strict improvement to me.

Could think about a concept on how we could correctly type the `status` returned from `getHealth` to be `200 | 206 | 503` instead of `number` which I can see adds some usefulness in this specific case.